deep_qa/common/params.py

```diff
@@ -44,34 +44,32 @@ def __init__(self, params: Dict[str, Any], history: str=""):
         self.history = history
 
     @overrides
-    def pop(self, key: str, default: Any=DEFAULT) -> Union["Params", Any]:
+    def pop(self, key: str, default: Any=DEFAULT):
         """
-        Performs the functionality associated with dict.pop(key) but with parameter
-        logging. This is required because pop_with_default may receive a default value
-        of None, which means we can't check for it not being passed.
+        Performs the functionality associated with dict.pop(key), along with checking for
+        returned dictionaries, replacing them with Param objects with an updated history.
         """
         if default is self.DEFAULT:
             value = self.params.pop(key)
         else:
             value = self.params.pop(key, default)
-        logger.param(key + " : " + str(value))
+        logger.param(self.history + " : " + key + " : " + str(value))
         return self.__check_is_dict(key, value)
 
     @overrides
-    def get(self, key: str, default: Any=DEFAULT) -> Union["Params", Any]:
+    def get(self, key: str, default: Any=DEFAULT):
```

Comment:
Oh, yeah, you're right.  I guess our tests should catch if you're using this wrong, anyway, because we check that the parameters are empty in `Trainer` - if you used `get` instead of `pop`, it'll get caught.